ebos/eclwriter.hh

```diff
@@ -250,14 +257,77 @@ public:
         eclOutputModule_.outputFipLog(miscSummaryData, regionData, isSubStep);
 
         // write output on I/O rank
+        std::vector<char> buffer;
         if (collectToIORank_.isIORank()) {
             const auto& eclState = simulator_.vanguard().eclState();
-            const auto& simConfig = eclState.getSimulationConfig();
 
             // Add TCPU
             if (totalCpuTime != 0.0)
                 miscSummaryData["TCPU"] = totalCpuTime;
 
+            const Opm::data::Wells& wellData = collectToIORank_.isParallel() ? collectToIORank_.globalWellData() : localWellData;
+
+            const std::map<std::pair<std::string, int>, double>& blockData
+                = collectToIORank_.isParallel()
+                ? collectToIORank_.globalBlockData()
+                : eclOutputModule_.getBlockData();
+
```

Comment:
This function is already long, so this part should be refactored into its own function (or multiple). I think a `broadcastSummaryState()` could hide the buffer and MPI calls from the rest of the function in a good way.

I love the long comment explaining the weirdness of the `if (reportStepNum > 0)` part, but I miss comments saying "what is going on here".

Nitpick: please avoid double blank lines inside functions. (Especially in files which have only a single blank between functions.